encodings/fastlanes/src/for/compute.rs

```diff
@@ -64,11 +71,28 @@ impl SliceFn for FoRArray {
     }
 }
 
+impl SearchSortedFn for FoRArray {
+    fn search_sorted(&self, value: &Scalar, side: SearchSortedSide) -> VortexResult<SearchResult> {
+        match_each_integer_ptype!(self.ptype(), |$P| {
+            let min: $P = self.reference().try_into().unwrap();
+            let shifted_min = min >> self.shift();
+            let unwrapped_value: $P = value.cast(self.dtype())?.try_into().unwrap();
+            let shifted_value: $P = unwrapped_value >> self.shift();
+            if shifted_value < shifted_min {
+                return Ok(SearchResult::NotFound(0));
```

Comment:
Is this short-cut worth it? Wouldn't the delegated search pretty much return this immediately?

Reply:
It is worth it since it lets you delegate. The fallback method will do scalar at recursively instead of search_sorted so you lose ability to delegate to any specialised implementation

Reply:
Oh, we have a real issue here. We need to change something about FORarrays

Reply:
I made an issue #400. For now I made FoR compressor bail in those cases. I think if we convert the dtypes we can make it work

Reply:
ok #401 handles this

Reply:
But after that merges I need to change the logic here to cast the translated scalar to unsinged ptype